test/e2e/tests/test_notebook_instance.py

```diff
@@ -0,0 +1,135 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# 	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""Integration tests for the SageMaker NotebookInstance API.
+"""
+
+import botocore
+import pytest
+import logging
+from typing import Dict
+
+from acktest.resources import random_suffix_name
+from acktest.k8s import resource as k8s
+from e2e import (
+    service_marker,
+    wait_for_status,
+    create_sagemaker_resource,
+    sagemaker_client,
+)
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.bootstrap_resources import get_bootstrap_resources
+from e2e.common import config as cfg
+from time import sleep
+import random
+
+RESOURCE_PLURAL = "notebookinstances"
+RESOURCE_PREFIX = "nb"
+RESOURCE_SPEC_FILE = "notebook_instance"
+
+
+@pytest.fixture(scope="function")
+def notebook_instance():
+    resource_name = RESOURCE_PREFIX + str(random.randint(0, 10000))
```

Comment:
Is there a reason for this str(random.randint(0, 10000)) and not using something like [this](https://github.com/aws-controllers-k8s/sagemaker-controller/blob/bb7eb30d6802ba6caf7e644f216e0c645bef527e/test/e2e/tests/test_trainingjob.py#L39)?

Reply:
Mentioned it in the other PR but it's because the Notebook does not support that kind of name.